hoss/coordinate_utilities.py

```diff
@@ -341,54 +332,69 @@ def create_dimension_arrays_from_coordinates(
         prefetch_dataset,
         longitude_coordinate.full_name_path,
     )
-    row_size, col_size = get_row_col_sizes_from_coordinate_datasets(
-        lat_arr,
-        lon_arr,
-    )
 
     row_indices, col_indices = get_valid_row_col_pairs(
         lat_arr, lon_arr, latitude_coordinate, longitude_coordinate
     )
 
-    y_dim = get_dimension_array_from_geo_points(
-        lat_arr, lon_arr, crs, row_indices, row_size, True
+    dim_order_is_y_x, row_dim_values = get_dimension_order_and_dim_values(
+        lat_arr, lon_arr, row_indices, crs, is_row=True
     )
+    dim_order, col_dim_values = get_dimension_order_and_dim_values(
+        lat_arr, lon_arr, col_indices, crs, is_row=False
+    )
+    if dim_order_is_y_x != dim_order:
+        raise InvalidCoordinateData("the order of dimensions do not match")
 
-    x_dim = get_dimension_array_from_geo_points(
-        lat_arr, lon_arr, crs, col_indices, col_size, False
+    row_size, col_size = get_row_col_sizes_from_coordinate_datasets(
+        lat_arr, lon_arr, dim_order_is_y_x
+    )
+
+    y_dim = get_1d_dim_array_data_from_dimvalues(
+        row_dim_values, np.transpose(row_indices)[0], row_size
+    )
+
+    x_dim = get_1d_dim_array_data_from_dimvalues(
+        col_dim_values, np.transpose(col_indices)[1], col_size
     )
 
     projected_y, projected_x = tuple(projected_dimension_names)
-    return {projected_y: y_dim, projected_x: x_dim}
 
+    if dim_order_is_y_x:
+        return {projected_y: y_dim, projected_x: x_dim}
+    return {projected_x: x_dim, projected_y: y_dim}
```

Comment:
I'm guessing this is formatted this way due to the no-else-return thing from `pylint` again?

Also, are these return values meaningfully different?

* They are both dictionaries
* Both have the same keys
* The keys map to the same values in both (e.g., `projected_x` -> `x_dim` and `projected_y` -> `y_dim`).

The only difference is the ordering of the dictionary keys - is that something that you are relying on elsewhere? (Python dictionary ordering is now safe to rely on, but traditionally, that's been a thing you shouldn't do)

If you don't rely on the ordering of the dictionary keys, then I would recommend consolidating this and removing the `if` branch here.

Reply:
It will affect the order of the range keys....
I will not change it for now. But I can revisit this in the next PR

It is used in spatial.py lines 262-265
     projected_y, projected_x = dimension_arrays.keys()
     if not set((projected_x, projected_y)).issubset(set(index_ranges.keys())):